src/compton2/Compton2.cc

```diff
@@ -0,0 +1,475 @@
+//----------------------------------*-C++-*-----------------------------------//
+/*!
+ * \file   compton2/Compton2.cc
+ * \author Andrew Till
+ * \date   11 May 2020
+ * \brief  Implementation file for compton interface
+ * \note   Copyright (C) 2017-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//----------------------------------------------------------------------------//
+
+// headers provided in draco:
+#include "compton2/Compton2.hh"
+#include "c4/C4_Functions.hh" //node()
+#include "c4/global.hh"
+#include "ds++/Assert.hh"
+
+#include <algorithm>
+#include <array>
+#include <cstring>
+#include <fstream>
+#include <iomanip> // RM
+#include <ios>     // RM
+#include <iostream>
+
+using UINT = size_t;
+using FP = double;
+using vec = std::vector<FP>;
+
+namespace rtt_compton2 {
+
+// xs is assumed to be monotonically increasing and unique
```

Comment:
Please add doxygen style comment block.
Did you intend for this to be a file-scope free function?
Would it make more sense to define these as template functions `<UINT,FP>` and omit the `#define` above?

Reply:
I'll dig around for one. Yes, the function is a helper function that doesn't need to be in the class, though I'm not sure who else would use it.

We could templatize on UINT and FP. Were you thinking just for the free functions or for the entire class?

Reply:
> I'll dig around for one. Yes, the function is a helper function that doesn't need to be in the class, though I'm not sure who else would use it.
> 
> We could templatize on UINT and FP. Were you thinking just for the free functions or for the entire class?

I was thinking for the entire class since you use these CPP macros in a way that a native C++ template could be used.  The template form is less error prone and the static analysis tools do a better job of checking.

Reply:
I need to think more about this. I'd like FP and UINT to be the same as the data types that are stored in the binary csk data file but then I'd like to return doubles and size_t or uint64_t's. Does that seem more reasonable?

Reply:
That makes perfect sense.  CSK should be able to publish the data types through a header or `config.h` or even through the cmake system. (`csk-config.cmake`).  You might even be able to use to C++'s [decltype](https://en.cppreference.com/w/cpp/language/decltype) to return the type of a data field defined by CSK.

```c++
class compton2 {
private:
  // DATA
  decltype(csk::fp_type) csk_data;
  double compton2_data;
...}
```

I'm not sure if that snippet is valid C++, but you get the idea.